cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/CruiseControlMetricsReporterSampler.java

```diff
@@ -224,6 +228,9 @@ public void configure(Map<String, ?> configs) {
       reconnectBackoffMs = String.valueOf(DEFAULT_RECONNECT_BACKOFF_MS);
     }
 
+    _acceptableMetricRecordProduceDelayMs = ACCEPTABLE_NETWORK_DELAY_MS + new CruiseControlMetricsReporterConfig(configs, false)
```

Comment:
I am wondering if the `ACCEPTABLE_NETWORK_DELAY_MS` would potentially be longer than the `linger.ms`? For example, what happens if getting free memory from buffer takes longer than `linger.ms`, but less than max block ms? should we set the `_acceptableMetricRecordProduceDelayMs` to `CRUISE_CONTROL_METRICS_REPORTER_INTERVAL_MS_CONFIG`?

Reply:
`ACCEPTABLE_NETWORK_DELAY_MS` and `linger.ms` are for different things.
`ACCEPTABLE_NETWORK_DELAY_MS` is to model the largest time message to transmit in network.
`linger.ms` is the largest time between producer's async send call and message get really sent out.

The reason we want to calculate `_acceptableMetricRecordProduceDelayMs` is to model the discrepancy between record's event time and append time at broker, and the above two are two contributing part.

Reply:
^ discussed offline -- i.e. `_acceptableMetricRecordProduceDelayMs` might potentially be max(`linger.ms`, `mac.block.ms`) + `ACCEPTABLE_NETWORK_DELAY_MS`.